tests/bluetooth/controller/ctrl_ull_dletime/src/main.c

```diff
@@ -0,0 +1,196 @@
+#include "dle_2m.h"
+#endif
+#if (!defined CONFIG_BT_CTLR_PHY_2M) && (defined CONFIG_BT_CTLR_PHY_CODED)
+#include "dle_coded.h"
+#endif
+#if (!defined CONFIG_BT_CTLR_PHY_2M) && (!defined CONFIG_BT_CTLR_PHY_CODED)
+#include "dle_none.h"
+#endif
+#else
+#include "dle_no_phy.h"
+#endif
+#include "dle_pkt_us.h"
+
+struct ll_conn *conn;
+
+
+
+
+static K_SEM_DEFINE(sem_prio_recv, 0, UINT_MAX);
+
+/*
+ * Helper functions
+ * mainly introduced to keep linelength below 80 chars
+ */
+static void helper_loop_dle_time(u16_t fex, u16_t f_set, u16_t i)
+{
+	u16_t rx_time, tx_time;
+
+#ifdef CONFIG_BT_CTLR_PHY
+	u16_t j;
+#endif
+#ifdef CONFIG_BT_CTLR_PHY
+	for (j = 0; j < nr_of_time_elements; j++) {
+		conn->default_tx_time = default_time[j];
+		dle_max_time_get(conn, &rx_time, &tx_time);
+		zassert_equal(tx_time,
+			      expected_tx_time[j][i][f_set][fex],
+			      "Failure in unit test, %d\n %d\n",
+			      tx_time, expected_tx_time[0][i][f_set][fex]);
+		zassert_equal(rx_time,
+			      expected_rx_time[j][i][f_set][fex],
+			      "Failure in unit test, %d\n %d\n",
+			      rx_time, expected_rx_time[0][i][f_set][fex]);
+
+	}
+#else
+	dle_max_time_get(conn, &rx_time, &tx_time);
+
+	zassert_equal(tx_time,
+		      expected_tx_time[0][i][f_set][fex],
+		      "Failure in unit test, got %d\nexpected %d\n",
+		      tx_time, expected_tx_time[0][i][f_set][fex]);
+	zassert_equal(rx_time,
+		      expected_rx_time[0][i][f_set][fex],
+		      "Failure in unit test, got %d\nexpected %d\n",
+		      rx_time, expected_rx_time[0][i][f_set][fex]);
+
+#endif
+}
+
+static void helper_loop_pkt_us(u16_t phy_counter)
+{
+	u16_t nr_octets = ARRAY_SIZE(octets);
+	u16_t nr_us = ARRAY_SIZE(expected_us);
+
+	u16_t counter;
+	u16_t phy;
+	u16_t calc_time;
+
+	zassert_equal(nr_octets, nr_us, "Failure in test routine\n");
+
+	phy = phy_to_test[phy_counter];
+
+	for (counter = 0; counter < nr_octets; counter++) {
+		calc_time = PKT_US(octets[counter], phy);
+
+		zassert_equal(calc_time,
+			      expected_us[counter][phy_counter],
+			      "Failure in unit test: got -%d- expected -%d-",
+			      calc_time, expected_us[counter][phy_counter]);
+	}
+}
+
+/*
+ * Unit test routines
+ */
+
+void test_conn_init(void)
+{
+	u16_t err_state;
+
+	ll_init(&sem_prio_recv);
+	err_state = init_reset();
+	zassert_equal(err_state, 0, NULL);
+
+	conn = ll_conn_acquire();
+	zassert_not_null(conn, NULL);
+}
+
+
+
+
+void test_dle_max_time_get(void)
+{
+	u16_t i, fex, f_set;
+
+
+	zassert_equal(nr_of_size_elements, OCTETS_TO_TEST,
+		      "Failure in test procedure\n");
+	zassert_equal(nr_of_features, FEATURES_TO_TEST,
+		      "Failure in test procedure\n");
+
+	conn->llcp_feature.features = 0x00;
+	conn->common.fex_valid = 0;	/* no feature exchange */
+
+	for (fex = 0; fex <= 1; fex++) {
+		conn->common.fex_valid = fex;
+
+		for (f_set = 0; f_set < nr_of_features; f_set++) {
+			conn->llcp_feature.features = feature[f_set];
```

Comment:
Seems like the definition if `i` should be moved into this innermost loop to follow the principle of declaring variables in the smaller possible scope.

Reply:
Ok, that seems to be the way to do it in zephyr